src/main/java/com/flippingutilities/jobs/TimeseriesFetcher.java

```diff
@@ -0,0 +1,98 @@
+package com.flippingutilities.jobs;
+
+import com.flippingutilities.controller.FlippingPlugin;
+import com.flippingutilities.model.CachedTimeseries;
+import com.flippingutilities.model.TimeseriesCacheKey;
+import com.flippingutilities.model.Timestep;
+import com.flippingutilities.model.TimeseriesResponse;
+import com.google.gson.JsonSyntaxException;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
+
+@Slf4j
+@Singleton
+public class TimeseriesFetcher {
+    private static final String TIMESERIES_API_URL = "https://prices.runescape.wiki/api/v1/osrs/timeseries";
+    private static final String QUERY_PARAM_TIMESTEP = "timestep";
+    private static final String QUERY_PARAM_ID = "id";
+    private static final String USER_AGENT_HEADER = "User-Agent";
+    private static final String USER_AGENT_VALUE = "FlippingUtilities";
```

Comment:
should also include the contact method, so - Anyny0 or discord.gg/flipping, but they know us so it's fine